src/compiler/checker.ts

```diff
@@ -330,6 +331,7 @@ module ts {
             let lastLocation: Node;
             let propertyWithInvalidInitializer: Node;
             let errorLocation = location;
+            let argumentsUsedInArrowFunction = false;

```

Comment:
I would make this more general. Have a variable that simply tracks whether you've reported an error.